src/overseer/store/jdbc.clj

```diff
@@ -0,0 +1,238 @@
+  (let [lock-version (query-lock-version db-spec job-id)
+        set-map' (assoc set-map :lock_version (inc lock-version))
+        where-clause [(format "id = ? AND lock_version = ? AND %s" (where-map->clause where-map)) job-id lock-version]]
+    (let [num-rows-updated (first (j/update! db-spec :overseer_jobs set-map' where-clause))]
+      (if (not= 1 num-rows-updated)
+        (throw (ex-info (str "Attempted to update stale job " (pr-str job-id)) {:type :stale-object}))
+        num-rows-updated))))
+
+(defn- direct-dependents
+  "Find the set of all Job IDs that directly depend on given `job-id`"
+  [db-spec job-id]
+  (->> (j/query db-spec ["SELECT job_id FROM overseer_dependencies WHERE dep_id = ?" job-id])
+       (map :job_id)
+       set))
+
+(defn dependents [db-spec job-id]
+  "Returns the set of all Job IDs that depend upon given `job-id`, both
+  directly and transitively"
+  (loop [all-dependents #{}
+         visited #{}
+         to-visit #{job-id}]
+    (if (empty? to-visit)
+      (set (map #(:job/id (query-job db-spec %)) all-dependents))
+      (let [job-id' (first to-visit)
+            dependents (direct-dependents db-spec job-id')
+            all-dependents' (set/union all-dependents dependents)
+            visited' (conj visited job-id')
+            to-visit' (set/difference
+                        (set/union to-visit dependents)
+                        visited')]
+        (recur all-dependents' visited' to-visit')))))
+
+(defn job-dep-jdbc-maps
+  "Return a seq of dependency rows to insert for a Graph<Job>"
+  [graph]
+  (let [nodes (loom.graph/nodes graph)
+        edges (loom.graph/edges graph)
+
+        dep-edges ; Basically recreate adjacency list {job [dep-jobs]}
+        (std/map-kv (partial map second) (group-by first edges))
+
+        job->dep-maps
+        (fn [job]
+          (when (contains? dep-edges job)
+            (->> (get dep-edges job)
+                 (map (fn [dep]
+                        {:job_id (:job/id job)
+                         :dep_id (:job/id dep)})))))]
+    (->> nodes
+         (mapcat job->dep-maps)
+         (filter identity))))
+
+(def default-table-opts
+  "Backend-specific system configuration options"
+  {:mysql
+   {:table-spec "ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin ROW_FORMAT=DYNAMIC"}})
+
+(defn install' [adapter db-spec]
+  (j/db-do-commands
+    db-spec
+    [(j/create-table-ddl
+       :overseer_jobs
+       [[:id "VARCHAR(64) PRIMARY KEY"]
+        [:type "VARCHAR(255)"]
+        [:args "VARCHAR(2048)"]
+        [:status "VARCHAR(255)"]
+        [:failure "VARCHAR(2048)"]
+        [:heartbeat :int]
+        [:lock_version "INT NOT NULL DEFAULT '0'"]]
+       (get default-table-opts adapter {}))
+     (j/create-table-ddl
+       :overseer_dependencies
+       [[:job_id "VARCHAR(64)"]
+        [:dep_id "VARCHAR(64)"]]
+       (get default-table-opts adapter {}))
+     "CREATE INDEX index_overseer_jobs_on_status ON overseer_jobs (status)"
```

Comment:
I don't think our current query patterns would benefit from an index on type, unless you're picturing out of band manual queries like `select * from jobs where type = 't1'`. I like the idea of timestamp columns; will add